### PR TITLE
feat/util(CL): Go client for CL positions

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

### Concentrated Liquidity Go Client

This Go-client allows connecting to an Osmosis chain via Ignite CLI and
setting up a concentrated liquidity pool with positions.

#### General Setup FAQ

- Update constants at the top of the file accordingly.
   * Make sure keyring is set up.
   * Client home is pointing to the right place.

#### LocalOsmosis Setup

Make sure that you run `localosmosis` in the background and have keys
added to your keyring with:
```bash
make localnet-keys
```

See `tests/localosmosis` for more info.

Also, ran `go work sync`

## Running

```bash
go run tests/cl-go-client/main.go
```

In the current state, it does the following:
- Queries status of the chain to make sure it's running.
- Queries pool with id 1. If does not exist, creates it
- Sets up one CL position

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

**Comment by p0mvn** [comment]
The diff is stemming from running `go work sync`

**Comment by p0mvn** [comment]
This now creates 1000 randomized positions from various accounts. This is r4r.  @czarcas7ic @stackman27 @mattverse  please review.

Example output:
```
2023/03/11 01:18:29 created position: amt0 0 amt1 18450071 liquidity 12456450078.028394745997828760
2023/03/11 01:18:29 creating position: pool id 1 accountName lo-test2 lowerTick 321 upperTick 1979 token0Desired 41778805uosmo tokenDesired1 74569258uion defaultMinAmount 0
2023/03/11 01:18:34 created position: amt0 41778805 amt1 0 liquidity 3263028984.671519136369081913
2023/03/11 01:18:34 creating position: pool id 1 accountName lo-test3 lowerTick -388 upperTick 1526 token0Desired 31163850uosmo tokenDesired1 46721379uion defaultMinAmount 0
2023/03/11 01:18:39 created position: amt0 31163850 amt1 30899416 liquidity 31163850.100580893702711779
2023/03/11 01:18:39 creating position: pool id 1 accountName lo-test3 lowerTick 818 upperTick 1387 token0Desired 23731422uosmo tokenDesired1 71854123uion defaultMinAmount 0
2023/03/11 01:18:44 created position: amt0 23731422 amt1 0 liquidity 1007462436300.942264359419860235
2023/03/11 01:18:44 creating position: pool id 1 accountName lo-test2 lowerTick 801 upperTick 893 token0Desired 93546637uosmo tokenDesired1 84230399uion defaultMinAmount 0
2023/03/11 01:18:49 created position: amt0 93546637 amt1 0 liquidity 4106496000697.315699447924584745
2023/03/11 01:18:49 creating position: pool id 1 accountName lo-test7 lowerTick 1256 upperTick 2048 token0Desired 58522529uosmo tokenDesired1 4122822uion defaultMinAmount 0
2023/03/11 01:18:54 created position: amt0 58522529 amt1 0 liquidity 573421454984345.792861106669436959
2023/03/11 01:18:54 creating position: pool id 1 accountName lo-test7 lowerTick 1657 upperTick 3254 token0Desired 55595706uosmo tokenDesired1 56123796uion defaultMinAmount 0
2023/03/11 01:18:59 created position: amt0 55595706 amt1 0 liquidity 120528618639555333.705084624731336203
2023/03/11 01:18:59 creating position: pool id 1 accountName lo-test5 lowerTick -116 upperTick 414 token0Desired 17471788uosmo tokenDesired1 37551529uion defaultMinAmount 0
2023/03/11 01:19:04 created position: amt0 17471788 amt1 12769423 liquidity 17541125.386178656501498144
2023/03/11 01:19:04 creating position: pool id 1 accountName lo-test6 lowerTick 2960 upperTick 3256 token0Desired 3844559uosmo tokenDesired1 78545335uion defaultMinAmount 0
2023/03/11 01:19:09 created position: amt0 3844559 amt1 0 liquidity 117523315996143806038833.260803714754525535
2023/03/11 01:19:09 creating position: pool id 1 accountName lo-test1 lowerTick 2603 upperTick 3087 token0Desired 84457147uosmo tokenDesired1 69878397uion defaultMinAmount 0
```

File: tests/cl-go-client/main.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		var (
+			// 1 to 9. These are localosmosis keyring test accounts with names such as:
+			// lo-test1
+			// lo-test2
+			// ...
+			randAccountNum = rand.Intn(8) + 1
+			accountName    = fmt.Sprintf("%s%d", accountNamePrefix, randAccountNum)
+			// minTick <= lowerTick <= upperTick
+			lowerTick = rand.Int63n(maxTick-minTick+1) + minTick
+			// lowerTick <= upperTick <= upperTick
```
Comment (stackman27) [inline tests/cl-go-client/main.go:94]:
```suggestion
			// lowerTick <= upperTick <= maxTick
```

**Review by stackman27 — COMMENTED** [review]
Just checked nits and have a quick question 

1. What is the difference between creating this goclient vs generating and running msgs through simulator?

**Comment by p0mvn** [comment]
> Just checked nits and have a quick question 
> 
> 
> 
> 1. What is the difference between creating this goclient vs generating and running msgs through simulator? 

Simulator runs as a Go test with only part of the system enabled.

On the other hand, this creates positions for the actual chain. The chain being localosmosis,testnet or main et depending on the config

**Comment by stackman27** [comment]
> > Just checked nits and have a quick question
> > 
> > 1. What is the difference between creating this goclient vs generating and running msgs through simulator?
> 
> They serve slightly different purposes.
> 
> Simulator runs as a Go test with only part of the system enabled.
> 
> On the other hand, Go client creates positions for the actual chain. The chain being localosmosis, testnet or mainnet depending on the config.
> 
> With this Go client, we want to setup a CL pool on a "real node" where there are many positions.

i see that makes sense

**Review by czarcas7ic — APPROVED** [review]
This LGTM, with the additions for sequence numbers on the next iteration. Nice work

**Pull request merged**